IdentityCore/src/parameters/MSIDInteractiveRequestParameters.m

```diff
@@ -72,14 +72,18 @@ - (NSOrderedSet *)allAuthorizeRequestScopes
 
 - (NSDictionary *)allAuthorizeRequestExtraParameters
 {
-    if (!self.extraAuthorizeURLQueryParameters && !self.extraURLQueryParameters)
+    NSMutableDictionary *authorizeParams = [[NSMutableDictionary alloc] initWithDictionary:self.appRequestMetadata];
+    
+    if (self.extraAuthorizeURLQueryParameters && self.extraAuthorizeURLQueryParameters.count > 0)
```

Comment:
you can just check for count, if it's nil it will still return 0

Reply:
it just makes it more explicit :p i'm fine either way.